docs/release-notes/flink-1.12.md

```diff
@@ -0,0 +1,169 @@
+
+Deprecated and methods without effect were removed: `ExecutionConfig#enable/disableSysoutLogging`, `ExecutionConfig#set/isFailTaskOnCheckpointError`.
+
+Removed `-q` flag from cli. The option had no effect. 
+
+#### Remove deprecated RuntimeContext#getAllAccumulators [FLINK-19032](https://issues.apache.org/jira/browse/FLINK-19032)
+
+The deprecated method `RuntimeContext#getAllAccumulators` was removed. Please use `RuntimeContext#getAccumulator` instead. 
+
+#### Deprecated CheckpointConfig#setPreferCheckpointForRecovery due to risk of data loss [FLINK-20441](https://issues.apache.org/jira/browse/FLINK-20441)
+
+The `CheckpointConfig#setPreferCheckpointForRecovery` method has been deprecated, because using checkpoints for recovery can lead to data loss.
+
+#### FLIP-134: Batch execution for the DataStream API
+
+- Allow explicitly configuring time behaviour on `KeyedStream.intervalJoin()` [FLINK-19032](https://issues.apache.org/jira/browse/FLINK-19032)
+
+  Before Flink 1.12 the `KeyedStream.intervalJoin()` operation was changing behavior based on the globally set Stream TimeCharacteristic. In Flink 1.12 we introduced explicit `inProcessingTime()` and `inEventTime()` methods on `IntervalJoin` and the join no longer changes behaviour based on the global characteristic. 
+
+- Deprecate `timeWindow()` operations in DataStream API [FLINK-19318](https://issues.apache.org/jira/browse/FLINK-19318)
+
+  In Flink 1.12 we deprecated the `timeWindow()` operations in the DataStream API. Please use `window(WindowAssigner)` with either a `TumblingEventTimeWindows`, `SlidingEventTimeWindows`, `TumblingProcessingTimeWindows`, or `SlidingProcessingTimeWindows`. For more information, see the deprecation description of `TimeCharacteristic`/`setStreamTimeCharacteristic`. 
+
+- Deprecate `StreamExecutionEnvironment.setStreamTimeCharacteristic()` and `TimeCharacteristic` [FLINK-19319](https://issues.apache.org/jira/browse/FLINK-19319)
+
+  In Flink 1.12 the default stream time characteristic has been changed to `EventTime`, thus you don't need to call this method for enabling event-time support anymore. Explicitly using processing-time windows and timers works in event-time mode. If you need to disable watermarks, please use `ExecutionConfig.setAutoWatermarkInterval(long)`. If you are using `IngestionTime`, please manually set an appropriate `WatermarkStrategy`. If you are using generic "time window" operations (for example `KeyedStream.timeWindow()` that change behaviour based on the time characteristic, please use equivalent operations that explicitly specify processing time or event time). 
+
+- Remove deprecated `DataStream#split` [FLINK-19083](https://issues.apache.org/jira/browse/FLINK-19083)
+
+  The `DataStream#split()` operation has been removed after being marked as deprecated for a couple of versions. Please use [Side Outputs]({% link dev/stream/side_output.md %})) instead.
```

Comment:
Removing `split` and `fold` is not related to the `Batch` execution. It's just an API cleanup similar to removing UdfAnalyzer.